crates/bevy_render/src/color/mod.rs

```diff
@@ -406,6 +408,237 @@ impl Color {
         self
     }
 
+    // Linear RGB
+
+    /// Get the red in linear rgb colorspace.
+    pub fn linear_rgb_r(&self) -> f32 {
+        match self.as_rgba_linear() {
+            Color::RgbaLinear { red, .. } => red,
+            _ => unreachable!(),
+        }
+    }
+
+    /// Get the green in linear rgb colorspace.
+    pub fn linear_rgb_g(&self) -> f32 {
+        match self.as_rgba_linear() {
+            Color::RgbaLinear { green, .. } => green,
+            _ => unreachable!(),
+        }
+    }
+
+    /// Get the blue in linear rgb colorspace.
+    pub fn linear_rgb_b(&self) -> f32 {
+        match self.as_rgba_linear() {
+            Color::RgbaLinear { blue, .. } => blue,
+            _ => unreachable!(),
+        }
+    }
+
+    /// Set the red in linear rgb colorspace.
+    pub fn set_linear_rgb_r(&mut self, r: f32) -> &mut Self {
+        *self = self.as_rgba_linear();
+        match self {
+            Color::RgbaLinear { red, .. } => *red = r,
+            _ => unreachable!(),
+        }
+        self
+    }
+
+    /// Returns this color with red set to a new value in linear rgb colorspace.
+    #[must_use]
+    pub fn with_linear_rgb_r(mut self, r: f32) -> Self {
+        self.set_linear_rgb_r(r);
+        self
+    }
+
+    /// Set the green in linear rgb colorspace.
+    pub fn set_linear_rgb_g(&mut self, g: f32) -> &mut Self {
+        *self = self.as_rgba_linear();
+        match self {
+            Color::RgbaLinear { green, .. } => *green = g,
+            _ => unreachable!(),
+        }
+        self
+    }
+
+    /// Returns this color with green set to a new value in linear rgb colorspace.
+    #[must_use]
+    pub fn with_linear_rgb_g(mut self, g: f32) -> Self {
+        self.set_linear_rgb_g(g);
+        self
+    }
+
+    /// Set the blue in linear rgb colorspace.
+    pub fn set_linear_rgb_b(&mut self, b: f32) -> &mut Self {
+        *self = self.as_rgba_linear();
+        match self {
+            Color::RgbaLinear { blue, .. } => *blue = b,
+            _ => unreachable!(),
+        }
+        self
+    }
+
+    /// Returns this color with blue set to a new value in linear rgb colorspace.
```

Comment:
```suggestion
    /// Returns this color with blue set to a new value in rgb linear colorspace.
```

(Same for other comments, too)